core/commands/filestore.go

```diff
@@ -0,0 +1,306 @@
+
+If one or more <obj> is specified only verify those specific objects,
+otherwise verify all objects.
+
+The output is:
+
+<status> <hash> <size> <path> <offset>
+
+Where <status> is one of:
+ok:       the block can be reconstructed
+changed:  the contents of the backing file have changed
+no-file:  the backing file could not be found
+error:    there was some other problem reading the file
+missing:  <obj> could not be found in the filestore
+ERROR:    internal error, most likely due to a corrupt database
+
+For ERROR entries the error will also be printed to stderr.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("obj", false, true, "Cid of objects to verify."),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		_, fs, err := getFilestore(req)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		args := req.Arguments()
+		if len(args) > 0 {
+			out := perKeyActionToChan(args, func(c *cid.Cid) *filestore.ListRes {
+				return filestore.Verify(fs, c)
+			}, req.Context())
+			res.SetOutput(out)
+		} else {
+			next, err := filestore.VerifyAll(fs)
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+			out := listResToChan(next, req.Context())
+			res.SetOutput(out)
+		}
+	},
+	Marshalers: cmds.MarshalerMap{
+		cmds.Text: func(res cmds.Response) (io.Reader, error) {
+			outChan, ok := res.Output().(<-chan interface{})
+			if !ok {
+				return nil, u.ErrCast()
+			}
+			res.SetOutput(nil)
+			for r0 := range outChan {
+				r := r0.(*filestore.ListRes)
+				if r.Status == filestore.StatusOtherError {
+					fmt.Fprintf(res.Stderr(), "%s\n", r.ErrorMsg)
+				}
+				fmt.Fprintf(res.Stdout(), "%s %s\n", r.Status.Format(), r.FormatLong())
+			}
+			return nil, nil
+		},
+	},
+	Type: filestore.ListRes{},
+}
+
+var dupsFileStore = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Print block both in filestore and non-filestore.",
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		_, fs, err := getFilestore(req)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		ch, err := fs.FileManager().AllKeysChan(req.Context())
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
+		out := make(chan interface{})
+		res.SetOutput((<-chan interface{})(out))
+
+		go func() {
+			defer close(out)
+			for cid := range ch {
+				have, err := fs.MainBlockstore().Has(cid)
+				if err != nil {
+					out <- &RefWrapper{Err: err.Error()}
+					return
+				}
+				if have {
+					out <- &RefWrapper{Ref: cid.String()}
+				}
+			}
+		}()
+	},
+	Marshalers: refsMarshallerMap,
+	Type:       RefWrapper{},
+}
+
+var rmFileStore = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Remove IPFS block(s) from just the filestore or blockstore.",
+		ShortDescription: `
+Remove blocks from either the filestore or the main blockstore.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("hash", true, true, "Bash58 encoded multihash of block(s) to remove."),
```

Comment:
not bash but Base58, but as it is CID then it should accept just CID. CID can have base descriptor.